packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -486,6 +486,59 @@ class UsersRepo extends AbstractRepo<"Users"> {
       LIMIT $1;
     `, [limit])
   }
+  
+   /**
+   * Returns a list of users to whom we want to send an email
+   * asking them to fill in the annual EA Forum user survey.
+   *
+   * This excludes deleted/deactivated users,
+   * flagged or purged or removed from queue users,
+   * users who unsubscribed from all site emails,
+   * users who were banned any time over the past 6 month period,
+   * users who haven't been to the site in the past 2 years,
+   * and users who have already been sent this email.
```

Comment:
I'd be interested in feedback on these criteria (in addition to checking for bugs in the query below), if you have any

Reply:
I've checked for bugs and I'm reasonably confident there aren't any in the current query. I think those criteria are good. You could maybe add negative karma too?

Reply:
Good idea, I'll add that!